src/commands/heroku.rs

```diff
@@ -390,6 +392,152 @@ pub fn restart_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandR
     Ok(())
 }
 
+#[command]
+#[num_args(3)]
+pub fn deploy_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandResult {
+    let app_name = args
+        .single::<String>()
+        .expect("You must include an app name");
+
+    let source_url = args
+        .single::<String>()
+        .expect("You must include a link to a tar file containing the code you want to deploy");
+
+    let app_version = args
+        .single::<String>()
+        .expect("You must include an app version");
+
+    let buildpack_params = heroku_rs::endpoints::builds::post::BuildpackParam {
+        url: String::from("https://github.com/emk/heroku-buildpack-rust"),
+        name: String::from("emk/rust"),
+    };
+
+    // Create a build
+    let build_create_response = heroku_client(ctx).request(&builds::BuildCreate {
+        app_id: app_name.clone(),
+        params: builds::BuildCreateParams {
+            buildpacks: Some(vec![buildpack_params]),
+            source_blob: builds::SourceBlobParam {
+                checksum: None,
+                url: source_url,
+                version: Some(app_version.clone()),
+            },
+        },
+    });
+
+    if build_create_response.is_err() {
+        msg.reply(
+            &ctx,
+            format!(
+                "An error occured when trying to build {}:\n{:?}",
+                &app_name,
+                build_create_response.err()
+            ),
+        )?;
+
+        return Ok(());
+    }
+
+    let build = build_create_response.unwrap();
+
+    msg.reply(&ctx, build_response(app_name.clone(), &build))?;
+
+    // Check status of the build
+    let mut build_info_response = heroku_client(ctx).request(&builds::BuildDetails {
+        app_id: app_name.clone(),
+        build_id: build.clone().id,
+    });
+
+    if build_info_response.is_err() {
+        msg.reply(
+            &ctx,
+            format!(
+                "An error occured when trying to get the status of build {} for {}",
+                &build.id, &app_name,
+            ),
+        )?;
+
+        return Ok(());
+    }
```

Comment:
This code is duplicated here and inside the while loop. Maybe you can do something like:

```rust
fn is_build_finished(app_name: &str, build_id:  u64) -> Result<bool, Error> {
    // ...
}

// ...

while !is_build_finished(&app_id, build.id)? {
    // ...
}
```

Reply:
Implemented it a little different, but got rid of the duplicated code